src/lib/utils/toc.ts

```diff
@@ -3,11 +3,12 @@ import type { SourceHeadingItem, ToCItem } from "@/lib/types"
 
 // RegEx patterns
 const customIdRegEx = /^.+(\s*\{#([^\}]+?)\}\s*)$/
-const emojiRegEx = /<Emoji [^/]+\/>/g
-const emojiUnicodeRegEx = /\\u{[0-9A-F]+?}/gi
-const escapeSlashRegEx = /\\(?=[+*?^$\\.[\]{}()|/#])/g
+const unicodeEmojiRegEx = /\\u{[0-9A-F]+}/gi
+const unicodeIntlRegEx = /\\u[0-9A-F]+/gi
 const compiledSourceHeadingRegEx =
-  /(?<=mdx\()"h([2-4])",e\(.+?id:"([^"]+)".+?"([^"]+)"/g
+  /mdx\("h([2-4])",e\(.+?id:"([^"]+)"}\)((,("([^"]+)"|mdx\([^\)]+\)))+)\)/g
+const mdxFncRegEx = /mdx\(.+?\)/g
+const stringRegEx = /"([^"]+)"/g
```

Comment:
`compiledSourceHeadingRegEx` will now match the syntax for any h2/3/4 inside the compiled source. This will specifically handle grabbing the heading level, the id, and the _entire_ rest which contains the label.

This label is then parsed further to remove any instances of `mdx(...)` using `mdxFcnRegEx`... This removes any Emoji's for example. The remainder is then parsed for text between quotes, and joins them together.

===

Example of compiled source syntax with an Emoji component in the middle of a heading:
```
mdx("h3", e({}, { id: "about-us" }), "About ", mdx(Emoji, { text: "\u{1F984}", mdxType: "Emoji" }), "the team")
```

This will parse to:
```json
{
  "depth": 3,
  "id": "about-us",
  "label": "About the team"
}
```